go/tasks/plugins/k8s/ray/config.go

```diff
@@ -1,11 +1,65 @@
 package ray
 
 import (
+	"fmt"
+	"io/ioutil"
+
 	pluginsConfig "github.com/flyteorg/flyteplugins/go/tasks/config"
+	"github.com/pkg/errors"
+	restclient "k8s.io/client-go/rest"
 )
 
 //go:generate pflags Config --default-var=defaultConfig
 
+type ClusterConfig struct {
```

Comment:
Does it makes sense to move the `ClusterConfig` and `Auth` structs, probably the `KubeClientConfig` func as well, over to the `go/tasks/pluginmachinery/k8s` directory?

Then adding this support to all k8s plugins would be as simple as adding the `ClusterConfig` to the plugin configuration and setting the `CustomKubeClient` on the `PluginEntry` during registration.

Reply:
That would be the best-case scenario. We tried narrowing the focus to ray plugin to make things fast. Ideally, specifying a custom kubeclient will be possible for all k8s plugins. I'll refactor `ClusterConfig` and `Auth` structs to be under `go/tasks/pluginmachinery/k8s`.

Reply:
PTAL